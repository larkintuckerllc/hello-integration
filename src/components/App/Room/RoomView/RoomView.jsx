import { PropTypes } from 'prop-types';
import React from 'react';

const Room = ({ on, toggleOn }) => (
  <div>
    <div id="room__light">{on ? 'On': 'Off'}</div>
    <button onClick={toggleOn}>Toggle</button>
  </div>
);

Room.propTypes = {
  on: PropTypes.bool.isRequired,
  toggleOn: PropTypes.func.isRequired,
};

export default Room;
