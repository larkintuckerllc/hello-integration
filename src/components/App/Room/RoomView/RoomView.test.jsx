import Enzyme, { shallow } from 'enzyme';
import enzymeAdapterReact16 from 'enzyme-adapter-react-16';
import React from 'react';
import RoomView from './RoomView';

Enzyme.configure({ adapter: new enzymeAdapterReact16() });

const getDefaultProps = () => ({
  on: false,
  toggleOn: jest.fn(),
});

describe('RoomView component', () => {
  it('shallow renders without crashing (off)', () => {
    const defaultProps = getDefaultProps();
    const wrapper = shallow(<RoomView {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('shallow renders without crashing (on)', () => {
    const { on, ...defaultProps } = getDefaultProps();
    const wrapper = shallow(<RoomView on={true} {...defaultProps} />);
    expect(wrapper).toMatchSnapshot();
  });
});