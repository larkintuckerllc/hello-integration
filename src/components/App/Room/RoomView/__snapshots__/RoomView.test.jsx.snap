// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomView component shallow renders without crashing (off) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Room
    on={false}
    toggleOn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="room__light"
>
          Off
</div>,
        <button
          onClick={[Function]}
>
          Toggle
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Off",
          "id": "room__light",
        },
        "ref": null,
        "rendered": "Off",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Toggle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Toggle",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="room__light"
>
            Off
</div>,
          <button
            onClick={[Function]}
>
            Toggle
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Off",
            "id": "room__light",
          },
          "ref": null,
          "rendered": "Off",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Toggle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Toggle",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`RoomView component shallow renders without crashing (on) 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Room
    on={true}
    toggleOn={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          id="room__light"
>
          On
</div>,
        <button
          onClick={[Function]}
>
          Toggle
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "On",
          "id": "room__light",
        },
        "ref": null,
        "rendered": "On",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Toggle",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "Toggle",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            id="room__light"
>
            On
</div>,
          <button
            onClick={[Function]}
>
            Toggle
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "On",
            "id": "room__light",
          },
          "ref": null,
          "rendered": "On",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Toggle",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Toggle",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
